presto-main/src/main/java/com/facebook/presto/execution/QueryManagerConfig.java

```diff
@@ -283,4 +287,32 @@ public QueryManagerConfig setQueryExecutionPolicy(String queryExecutionPolicy)
         this.queryExecutionPolicy = queryExecutionPolicy;
         return this;
     }
+
+    @Min(1)
+    public int getInitialMinWorkersRequired()
+    {
+        return initialMinWorkersRequired;
+    }
+
+    @Config("query-manager.initial-min-workers-required")
+    @ConfigDescription("Minimum number of workers that must announce themselves before the cluster is ready to accept queries")
+    public QueryManagerConfig setInitialMinWorkersRequired(int initialMinWorkersRequired)
+    {
+        this.initialMinWorkersRequired = initialMinWorkersRequired;
+        return this;
+    }
+
+    @NotNull
+    public Duration getInitializationTimeout()
+    {
+        return initializationTimeout;
+    }
+
+    @Config("query-manager.initialization-timeout")
+    @ConfigDescription("Maximum amount of time the cluster can remain in the initializing state (i.e. unable to accept queries)")
```

Comment:
"After this time, the cluster will accept queries even if the minimum required workers are not available"